### PR TITLE
feat(conditionBuilder): tree variant initial implmentation

### PR DESCRIPTION
**Pull request by amal-k-joy**
Contributes #4916 

This the initial PR with the implementation of tree variant of condition builder.

#### What did you change?

- Tree variant structure. Tree variant basic structure is different from sentence variant. It follow a tree-grid pattern.
- Support for multiple condition groups 
- Support for adding subgroups within a group
- Separator section between groups (Group seperator)
- CSS for group colours logic.
- Support to show preview handlers for add new condition, new subgroup, new group
- Support for more stories

#### How did you test and verify your work?
Local storybook

### CODE REVIEW

File: packages/ibm-products-styles/src/components/ConditionBuilder/styles/_conditionBuilderCondition.scss [hunk 1]
```diff
@@ -14,12 +16,22 @@ $block-class: #{c4p-settings.$pkg-prefix}--condition-builder;
   display: inline-flex;
 }
 
-[role='row']:focus {
-  @include focus-outline.focus-outline('outline');
+// [role='row']:focus {
+//   @include focus-outline.focus-outline('outline');
```
Comment (makafsal) [inline packages/ibm-products-styles/src/components/ConditionBuilder/styles/_conditionBuilderCondition.scss:20]: Better remove commented code all across the files.

File: packages/ibm-products-styles/src/components/ConditionBuilder/styles/_conditionBuilderCondition.scss [hunk 2]
```diff
@@ -44,3 +56,15 @@ $block-class: #{c4p-settings.$pkg-prefix}--condition-builder;
   text-overflow: ellipsis;
   text-wrap: nowrap;
 }
+.#{$block-class}__group-preview {
+  opacity: 0.5;
+  pointer-events: none;
+}
+.#{$block-class}__connector-disabled {
+  display: flex;
+  min-width: 4rem;
```
Comment (makafsal) [inline packages/ibm-products-styles/src/components/ConditionBuilder/styles/_conditionBuilderCondition.scss:65]: Is it possible to replace it with a Carbon token?

**Review by makafsal — DISMISSED** [review]

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js [hunk 3]
```diff
@@ -74,46 +123,181 @@ const ConditionGroupBuilder = ({
...
+  const onConnectorOperatorChange = (op) => {
+    onChange({
+      ...group,
+      groupOperator: op,
+    });
   };
-  return (
-    <div className={`${className}  eachGroup`}>
+
+  const getSentenceVariant = () => {
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js:152]: in order to cut back on file length and improve readability i would move this and `getTreeVariant` into their own separate component files.

File: packages/ibm-products/src/components/ConditionBuilder/ConditionPreview/ConditionPreview.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+} from '../ConditionBuilderContext/DataConfigs';
+import { ConditionBuilderItem } from '../ConditionBuilderItem/ConditionBuilderItem';
+import ConditionConnector from '../ConditionBuilderConnector/ConditionConnector';
+const ConditionPreview = ({ previewType }) => {
+  const getConditionSection = () => {
+    return (
+      <div>
+        <ConditionBuilderItem
+          label={translateWithId('property')}
+        ></ConditionBuilderItem>
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/ConditionBuilder/ConditionPreview/ConditionPreview.js:15]: self closing notation preferred `<ConditionBuilderItem {...} />

File: packages/ibm-products/src/components/ConditionBuilder/ConditionPreview/ConditionPreview.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        </>
+      )}
+
+      {previewType == 'subGroup' && (
+        <div className={`${blockClass}__group-preview`}>
+          <div
+            className={`${blockClass}__condition-block conditionBlockWrapper  ${blockClass}__gap`}
+          >
+            <ConditionBuilderItem
+              label={'and'}
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/ConditionBuilder/ConditionPreview/ConditionPreview.js:58]: i see a few different hard coded strings here

**Review by davidmenendez — CHANGES_REQUESTED** [review]
looking good! just a few comments

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js (hunk 3 above)
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js:152]: Hey,
That makes sense and thought in the same way at first. But if you notice , most of the functionalities and handler definitions are same for both variant, except some of the JSX structure. So I thought separating into own files will make it more complex and duplicate as we need to passdown all handlers and call back. 
What do you think?

File: packages/ibm-products-styles/src/components/ConditionBuilder/styles/_conditionBuilderCondition.scss (hunk 1 above)
Comment (amal-k-joy) [reply packages/ibm-products-styles/src/components/ConditionBuilder/styles/_conditionBuilderCondition.scss]: resolved as advised

File: packages/ibm-products-styles/src/components/ConditionBuilder/styles/_conditionBuilderCondition.scss (hunk 2 above)
Comment (amal-k-joy) [reply packages/ibm-products-styles/src/components/ConditionBuilder/styles/_conditionBuilderCondition.scss]: resolved as advised

File: packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderAdd/ConditionBuilderAdd.js [hunk 6]
```diff
@@ -1,36 +1,134 @@
...
+          label={translateWithId('add-condition')}
+          wrapperProps={
+            enableSubGroup
+              ? {
+                  role: 'gridcell',
+                  'aria-label': translateWithId('add-condition'),
+                }
+              : {}
+          }
+          onMouseEnter={showConditionSubGroupPreviewHandler}
```
[the remark this answers is not in the document]
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderAdd/ConditionBuilderAdd.js]: made changes as advised

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js [hunk 7]
```diff
@@ -66,6 +81,40 @@ const ConditionGroupBuilder = ({
     });
   };
 
+  const handleFocusOnClose = (e) => {
+    let previousClose = e.currentTarget
```
[the remark this answers is not in the document]
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js]: made changes as advised

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js [hunk 8]
```diff
@@ -74,46 +123,181 @@ const ConditionGroupBuilder = ({
...
+                    onChangeHandler(updatedCondition, conditionIndex);
+                  }}
+                  onRemove={(e) => {
+                    onRemoveHandler(eachCondition.id, e);
+                  }}
+                  conditionBuilderRef={conditionBuilderRef}
+                />
+              )}
+              {/* rendering each condition block */}
+              {!eachCondition.conditions && (
```
[the remark this answers is not in the document]
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js]: made changes as advised

File: packages/ibm-products/src/components/ConditionBuilder/ConditionPreview/ConditionPreview.js (hunk 5 above)
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionPreview/ConditionPreview.js]: made changes as advised

File: packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderAdd/ConditionBuilderAdd.js [hunk 9]
```diff
@@ -1,36 +1,134 @@
...
+        }
       />
+      {isAddSubgroup && enableSubGroup && (
+        <ConditionBuilderButton
+          renderIcon={TextNewLine}
+          onClick={addConditionSubGroupHandler}
+          className={`${blockClass}__add-condition-group ${blockClass}__gap-left`}
+          hideLabel
+          label={translateWithId('add-condition')}
+          wrapperProps={
```
[the remark this answers is not in the document]
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderAdd/ConditionBuilderAdd.js]: made changes as advised

**Comment by amal-k-joy** [comment]
> looking good! just a few comments



> looking good! just a few comments



> looking good! just a few comments

Thanks for the review. I have resolved your comments

File: packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js (hunk 3 above)
Comment (davidmenendez) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js:152]: ok this should be fine for now. let's keep an eye on it

**Review by davidmenendez — APPROVED** [review]

**Review by makafsal — APPROVED** [review]

**Comment by amal-k-joy** [comment]
@all-contributors please add @amal-k-joy for code

**Pull request merged**